src/proficiency.cpp

```diff
@@ -332,6 +372,22 @@ std::vector<proficiency_id> proficiency_set::learning_profs() const
     return ret;
 }
 
+float proficiency_set::get_proficiency_bonus( const std::string &category,
+        proficiency_bonus_type prof_bonus ) const
+{
+    float stat_bonus = 0;
+
+    for( const proficiency_id &knows : known ) {
+        std::vector<proficiency_bonus> prof_bonuses = knows->get_bonuses( category );
```

Comment:
I think you're probably pretty good now, but I don't have the time to do a review at the moment.
Other than saying that you're copying this, and shouldn't be.
```suggestion
        const std::vector<proficiency_bonus> &prof_bonuses = knows->get_bonuses( category );
```